js/src/index.js

```diff
@@ -169,6 +170,21 @@ const registerPluginAdminPages = () => {
 			},
 		];
 
+		// When there is no connected MC account, restrict navigation to Ads-related and core onboarding routes only.
+		const hasMC = Boolean( glaData?.mcSetupComplete );
```

Comment:
@ankitrox Let's use a more descriptive name here or even use `glaData.mcSetupComplete` directly since we are using it once.

```suggestion
		const hasCompletedMcSetup = Boolean( glaData?.mcSetupComplete );
```
Also glaData.mcSetupComplete will be always defined, so no need for the optional chaining.